Create Manage Comments Form

alexisljn opened the pull request.

alexisljn committed:
Create Manage Comments Form

alexisljn committed:
Push To Work On Another Computer

alexisljn committed:
Modification Of Comment Model and others

Gii Model Generator added by itself a line in end file.
The edited date is now a nullable field.

alexisljn committed:
Create Logic For Post Comment Feature

Handling data from form to save it in DB.

alexisljn committed:
Change Approach To Find Related Proposal

Now it uses the $_SERVER['referrer'] instead of
an hidden input. That way we sure that nobody
can modify the proposal id.

alexisljn committed:
Start Comment Editing Security

Based on comment timestamp and supposed comment id
given by hidden input of the form.

alexisljn committed:
Change Of Comment Model

edited_date is now not nullable

alexisljn committed:
Change Of Fixtures

Comment edited date is now populated by comment date.

alexisljn committed:
Modification of Fixtures

edited date = date

alexisljn committed:
Finalize Add/Edit Comment Feature

Cancel link in view
changes in controller

alexisljn committed:
Cleaning JQuery code

2 selectors in 1 line.

alexisljn committed:
Edit Fixtures

Due to an error.

EmileCalixte commented on controllers/DevController.php:
```diff
@@ -102,6 +102,7 @@ public function actionFixtures()
                 $comment->author_id = $usersWhoCanPostComment[random_int(0, count($usersWhoCanPostComment)-1)]->id;
                 $comment->content = self::getLorem(10, 50);
                 $comment->date = (\DateTime::createFromFormat('U', random_int($minDate, $maxDate)))->format('Y-m-d H:i:s');
+                $comment->edited_date = $comment->date;
```
Why not null ?

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -570,4 +572,91 @@ private function removeExistingUploadedFile(string $filepath)
+
+        if ($model->load(Yii::$app->request->post()) && $model->validate()) {
+            $referrer = explode('/', Yii::$app->request->referrer);
+            $proposalId = $referrer[count($referrer) - 1];
```
I think it could be better (more understandable) to subit form to /proposals/post-comment/[proposal id] or just have a hidden input with the proposal ID. In all cases, you have to check if the user who is posting a comment is allowed to do it.

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -570,4 +572,91 @@ private function removeExistingUploadedFile(string $filepath)
+        $comment->content = $commentInput;
+        $comment->author_id = MainController::getCurrentUser()->id;
+        $comment->date = Util::getDateTimeFormattedForDatabase(new \DateTime());
+        $comment->edited_date = $comment->date;
```
Why not null ? I think it's better for a non-edited comment to have a null edited_date

alexisljn committed:
Change Models and Fixtures

Edited date is null by default.

alexisljn committed:
Change Method saveComment and View

To display edited date based on the conditional state
of the datetime, null or not.

alexisljn committed:
Change saveEditedComment

Add of verification

alexisljn committed:
Remove Referrer From Logic

Replaced by get param and orm logic.

alexisljn committed:
Clean useless dd()
